src/index.ts

```diff
@@ -204,8 +206,7 @@ function preactPlugin({
 			config = resolvedConfig;
 			devToolsEnabled =
 				devToolsEnabled ?? (!config.isProduction || devtoolsInProd);
-			useBabel =
-				!config.isProduction || devToolsEnabled || typeof babel !== "undefined";
+			useBabel ||= !config.isProduction || !!devToolsEnabled;
```

Comment:
```suggestion
			useBabel ||= !config.isProduction || devToolsEnabled;
```

It's already a boolean.